src/TableFunctions/TableFunctionIceberg.cpp

```diff
@@ -0,0 +1,169 @@
+#include "config.h"
+
+#if USE_AWS_S3
+
+#    include <filesystem>
+#    include <Access/Common/AccessFlags.h>
+#    include <Formats/FormatFactory.h>
+#    include <IO/S3Common.h>
+#    include <Interpreters/Context.h>
+#    include <Interpreters/evaluateConstantExpression.h>
+#    include <Interpreters/parseColumnsListForTableFunction.h>
+#    include <Parsers/ASTLiteral.h>
+#    include <Storages/StorageIceberg.h>
+#    include <Storages/StorageURL.h>
+#    include <Storages/checkAndGetLiteralArgument.h>
+#    include <TableFunctions/TableFunctionFactory.h>
+#    include <TableFunctions/TableFunctionIceberg.h>
+#    include "registerTableFunctions.h"
+
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int NUMBER_OF_ARGUMENTS_DOESNT_MATCH;
+}
+
+
+void TableFunctionIceberg::parseArgumentsImpl(
```

Comment:
let's reuse code from s3 table function as it is the same here, for example you can inherit from it.

Reply:
It's not totally same here, If we don't specify format for it, we can not get format from it and should use `Parquet` as default file format.